02_configure_host.sh

```diff
@@ -48,52 +48,59 @@ EOF
     virsh pool-autostart default
 fi
 
-if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
-    # Adding an IP address in the libvirt definition for this network results in
-    # dnsmasq being run, we don't want that as we have our own dnsmasq, so set
-    # the IP address here
-    if [ ! -e /etc/sysconfig/network-scripts/ifcfg-provisioning ] ; then
-        echo -e "DEVICE=provisioning\nTYPE=Bridge\nONBOOT=yes\nNM_CONTROLLED=no\nBOOTPROTO=static\nIPADDR=172.22.0.1\nNETMASK=255.255.255.0" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-provisioning
-    fi
-    sudo ifdown provisioning || true
-    sudo ifup provisioning
-
-    # Need to pass the provision interface for bare metal
-    if [ "$PRO_IF" ]; then
-        echo -e "DEVICE=$PRO_IF\nTYPE=Ethernet\nONBOOT=yes\nNM_CONTROLLED=no\nBRIDGE=provisioning" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-$PRO_IF
-        sudo ifdown $PRO_IF || true
-        sudo ifup $PRO_IF
-    fi
-fi
-
-if [ "$MANAGE_INT_BRIDGE" == "y" ]; then
-    # Create the baremetal bridge
-    if [ ! -e /etc/sysconfig/network-scripts/ifcfg-baremetal ] ; then
-        echo -e "DEVICE=baremetal\nTYPE=Bridge\nONBOOT=yes\nNM_CONTROLLED=no" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-baremetal
-    fi
-    sudo ifdown baremetal || true
-    sudo ifup baremetal
-
-    # Add the internal interface to it if requests, this may also be the interface providing
-    # external access so we need to make sure we maintain dhcp config if its available
-    if [ "$INT_IF" ]; then
-        echo -e "DEVICE=$INT_IF\nTYPE=Ethernet\nONBOOT=yes\nNM_CONTROLLED=no\nBRIDGE=baremetal" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-$INT_IF
-        if sudo nmap --script broadcast-dhcp-discover -e $INT_IF | grep "IP Offered" ; then
-            echo -e "\nBOOTPROTO=dhcp\n" | sudo tee -a /etc/sysconfig/network-scripts/ifcfg-baremetal
-            sudo systemctl restart network
-        else
-           sudo systemctl restart network
-        fi
-    fi
-fi
-
-# restart the libvirt network so it applies an ip to the bridge
-if [ "$MANAGE_BR_BRIDGE" == "y" ] ; then
-    sudo virsh net-destroy baremetal
-    sudo virsh net-start baremetal
-    if [ "$INT_IF" ]; then #Need to bring UP the NIC after destroying the libvirt network
-        sudo ifup $INT_IF
-    fi
+OS=$(uname -a)
+if [[ $OS == *Ubuntu* ]]; then
+  # source ubuntu_bridge_network_configuration.sh
+  source ubuntu_bridge_network_configuration.sh
```

Comment:
The Centos part could then also be split in its own script

Reply:
I also want to discuss with you guys about this. I tried to make as less change as possible, so it will be more easier for maintainer to review our code. For that reason, I tried to avoid changing to what they have done, and just adding new code.
What do you think about this ? Should we also modify what they have done to make the code easier to read ? Or maybe we can do it in the another pull request, so keep this pull request as familiar with the maintainer as possible ?

Reply:
I would say that we should not do too much changes. We need to check how our changes are accepted in the community. Let's hope pull request generates discussion in the community. I think we are not offering "ultimate" solution, but contribution dev-env for further development?